### PR TITLE
aalhommada reactjs week1

### PR DESCRIPTION
**Pull request by aalhommada**

### CODE REVIEW

File: homework-submission/week1/app.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+const Header = props => {
+  return <h3 className="text-center text-danger mt-4">{props.title}</h3>;
+};
+
+const StaticList = ({ done, description, deadline }) => {
```
Comment (karamad10) [inline homework-submission/week1/app.js]: done is destructured here but never used, only destructure props that you need.

File: homework-submission/week1/app.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      {props.done}
+      <span>
+        <input type="checkbox" onChange={props.handleComplete} />
+      </span>
+    </li>
+  );
+};
+
+class App extends React.Component {
+  constructor(props) {
```
Comment (karamad10) [inline homework-submission/week1/app.js]: props is also never used , here is a nice article about the difference between super() and super(props) 
https://overreacted.io/why-do-we-write-super-props/

File: homework-submission/week1/app.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    });
+
+    this.setState({
+      todos,
+    });
+  };
+
+  render() {
+    return (
+      <div className="container ">
```
Comment (karamad10) [inline homework-submission/week1/app.js]: extra space in the class name :)

File: homework-submission/week1/app.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        <div className="row ">
```
Comment (karamad10) [inline homework-submission/week1/app.js]: extra space in the class name :)

File: homework-submission/week1/app.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          <div className="col-10 col-md-8 mx-auto">
+            <div>
+              <Header title="Static List" />
+              <ul className="list-group">
+                <StaticList description="Get out of bed" deadline="Wed Sep 13 2017" />
+                <StaticList description="Brush teeth" deadline="Thu Sep 14 2017" />
+                <StaticList description="Eat breakfast" deadline="Fri Sep 15 2017" />
```
Comment (karamad10) [inline homework-submission/week1/app.js]: the name StaticList indicates a list but this is a list item, rename this component to make it more clear for anyone that reads your code,  maybe `StaticListItem` ?

File: homework-submission/week1/app.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+              <Header title="Dynamic List" />
+              <ul className="list-group">
+                {this.state.todos.map(todo => (
+                  <DynamicList
+                    key={todo.id}
+                    description={todo.description}
+                    deadline={todo.deadline}
+                    done={todo.done}
+                    handleComplete={this.handleComplete.bind(this, todo.id)}
+                  />
```
Comment (karamad10) [inline homework-submission/week1/app.js]: there is a few `todo.` here, maybe use a nice shortcut to repeat the same code less?

File: homework-submission/week1/app.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                <StaticList description="Brush teeth" deadline="Thu Sep 14 2017" />
+                <StaticList description="Eat breakfast" deadline="Fri Sep 15 2017" />
+              </ul>
+            </div>
+
+            <div>
+              <Header title="Dynamic List" />
+              <ul className="list-group">
+                {this.state.todos.map(todo => (
+                  <DynamicList
```
Comment (karamad10) [inline homework-submission/week1/app.js]: same here , you're returning a list item so maybe DynamicListItem

**Comment by karamad10** [comment]
hey @aalhommada , very good job, i left some improvement comments, also even though you didn't use create-react-app , you can still move your lists to a separate file or files like we did in class

File: homework-submission/week1/app.js (hunk 6 above)
Comment (aalhommada) [reply homework-submission/week1/app.js]: Thanks @karamad10  for your notes, 
I did not know exactly how to deconstructure it here ?!

**Comment by aalhommada** [comment]
Thanks @karamad10 , I modify the code as requested.

File: homework-submission/week1/app.js (hunk 6 above)
Comment (ArcoMul) [reply homework-submission/week1/app.js]: Sorry for the late reply, but destructering is something important to know. Read more here: https://wesbos.com/destructuring-objects/

Comment (aalhommada) [reply homework-submission/week1/app.js]: hi @ArcoMul , 

I know whats mean destructure her  and i can write it like this :
const {id. description, deadline, done } = todo;
but the problem that I face her: where should I put it, every time I try the code broken, could you please give me a hint?

Comment (aalhommada) [reply homework-submission/week1/app.js]: Hi @ArcoMul , 
I did it, I could not understand the question at the beginning .but now it is okay.

Comment (ArcoMul) [reply homework-submission/week1/app.js]: Superrr! :clap:

**Pull request closed without merging**